### DISCUSSION TITLE
Is there a cancellable nation-removes-town event thrown from the NewDay code?

### DISCUSSION THREAD

**Discussion by Fe1son** [TownyAdvanced/Towny#6516] [Q&A]
Hello, I need to disable the kick of a town from a nation during newday, I found different events, I don’t know which one to use and how.  NationRemoveTownEvent not canceled, NationPreTownKickEvent only when the leader of the nation kick the town. What event to use then? Can I please have a code example?

**Comment by LlmDl** [comment]
I took a look and you're right, the only way to stop a nation losing a town during the new day code is to listen for PreTownPaysNationTaxEvent and cancel it if the tax would not be paid.

**Reply by LlmDl** [reply]
```java
if (event.getTown().getAccount().getHoldingBalance() < event.getTax()) 
    event.setCancelled(true);
```

**Reply by Fe1son** [reply]
How can I also cancel the payment for the town upkeep?

**Reply by LlmDl** [reply]
https://github.com/TownyAdvanced/Towny#a-special-message-from-llmdl-precedes-the-normal-readme-

TownUpkeepCalculationEvent
